tests/src/python/test_qgsproviderconnection_postgres.py

```diff
@@ -395,6 +395,60 @@ def test_zm(self):
         table_info = conn.table('qgis_test', 'gh_43268_test_zm')
         self.assertEqual(sorted([QgsWkbTypes.displayString(col.wkbType) for col in table_info.geometryColumnTypes()]), ['LineStringZ', 'PointZ', 'PolygonZ'])
 
+    def test_table_scan(self):
+        """Test that with use estimated metadata disabled all geometry column
+        types can be identified, test for GH #43186 """
+
+        md = QgsProviderRegistry.instance().providerMetadata('postgres')
+        uri = QgsDataSourceUri(self.uri)
+        conn = md.createConnection(uri.uri(), {'estimatedMetadata': True})
+
+        sql = """
+        DROP TABLE IF EXISTS qgis_test.geometry_table_with_multiple_types;
+        CREATE TABLE qgis_test.geometry_table_with_multiple_types (
+            id SERIAL PRIMARY KEY,
+            geom geometry(Geometry,4326)
+        );
+        """
+
+        conn.executeSql(sql)
+
+        for i in range(110):
+            sql = "INSERT INTO qgis_test.geometry_table_with_multiple_types (geom) VALUES (ST_GeomFromText('point(9 45)', 4326));"
+            conn.executeSql(sql)
+
+        for i in range(10):
+            sql = "INSERT INTO qgis_test.geometry_table_with_multiple_types (geom) VALUES (ST_GeomFromText('linestring(9 45, 10 46)', 4326));"
+            conn.executeSql(sql)
+
+        table = conn.table('qgis_test', 'geometry_table_with_multiple_types')
+
+        self.assertEqual(len(table.geometryColumnTypes()), 1)
```

Comment:
```suggestion
        self.assertIn(len(table.geometryColumnTypes()), (1, 2))
```
the result depends on the order in which postgres will return the values

Reply:
Actually no: I'm checking the number of geometry types here.

Reply:
I meant that the number of geometry types in this place depends on the order in which the inner query will return values from the test table.
for example:
https://github.com/qgis/QGIS/blob/e7ccf1fcc6adace4f3764ea4c9772791bd2f36d6/src/providers/postgres/qgspostgresconn.cpp#L2004-L2008
If the Postgres query planner decides to return the "first" 100 records from the table, then everything will be fine, but if the "last" - test fails. In general, you can't predict exactly which 100 records from the test table will be used to count the number of geometry types.
But in fact, these are trifles, do not pay attention.